### PR TITLE
New API endpoint to fetch dataset directory

### PR DESCRIPTION
**Pull request by rahit**
New API for direcotory listing in homepage as described in #77 

## Description
Introducing new API endpoint which will be used to retrieve images from data directory.

## Reference to official issue
See issue #77 

## How Has This Been Tested?
This modification will effect homepage initial dataload. Implementation of the API is yet to be made.

## Updates of Acceptance Criteria
- [x] Update routes
- [x] Declare view (Not implementation)
- [x] Update index.html API call

## CLA
- [x] I have signed the CLA; if other committers are in the commit history, they have signed the CLA as well

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -63,5 +64,6 @@ prediction/data/*
 # ignore data
 data/
 
+
 # ignore Sphinx apidoc
-_apidoc_*
+_apidoc_*
```
Comment (lamby) [inline .gitignore]: ?

File: interface/backend/api/urls.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
     CandidateViewSet,
     NoduleViewSet,
     ImageSeriesViewSet,
+    ImageSrcApiView
```
Comment (lamby) [inline interface/backend/api/urls.py]: Please add a trailing comma to avoid VCS noise for the next addition! :)

**Comment by lamby** [comment]
Hey, thanks for this! Please rebase and — whilst you're at it — address those tiny niggling things I brought up :)

**Comment by rahit** [comment]
@lamby Resolved conflicts and modified as per your suggestion. I have mistakenly used master of forked repo for the pull request. I will do it in new branch for future changes. :)

File: .gitignore [hunk 3]
```diff
@@ -62,3 +63,7 @@ prediction/data/*
 
 # ignore data
 data/
+
+
+# ignore Sphinx apidoc
+_apidoc_*
```
Comment (isms) [inline .gitignore]: Please revert this change - see #80 (in particular https://github.com/concept-to-clinic/concept-to-clinic/pull/80/files)

File: interface/backend/api/urls.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
     CandidateViewSet,
     NoduleViewSet,
     ImageSeriesViewSet,
+    ImageSrcApiView,
```
Comment (isms) [inline interface/backend/api/urls.py]: Can we change this to `ImageAvailableApiView` or something more descriptive?

File: interface/backend/api/urls.py [hunk 5]
```diff
@@ -19,5 +20,6 @@
 
 urlpatterns = [
     url(r'^', include(router.urls)),
+    url(r'^image_src/$', ImageSrcApiView.as_view()),
```
Comment (isms) [inline interface/backend/api/urls.py]: As above, how about `^images/available$`?

File: interface/backend/api/views.py [hunk 6]
```diff
@@ -27,3 +28,14 @@ class NoduleViewSet(viewsets.ModelViewSet):
...
+
+class ImageSrcApiView(APIView): 
+    """ 
+    View list of images from dataset directory 
+    """ 
+    def get(self, request): 
+        """ 
+        Return a list of files and folders in dataset
+        """ 
+        pass
```
Comment (isms) [inline interface/backend/api/views.py]: Are you planning to implement or leave this stubbed out? Stub is acceptable if that's the plan, we'll just create another issue for actually implement this.

**Comment by lamby** [comment]
Hi @rahit 
Do you need any help or clarification with the review feedback? :)

File: interface/backend/api/views.py (hunk 6 above)
Comment (rahit) [reply interface/backend/api/views.py]: We can create a new issue for this.

**Comment by rahit** [comment]
Thank you @lamby for the suggestions. I could not get the chance to work on the modifications. I would like to let you know that your feedback is clear to me and I will get back asap.

**Comment by lamby** [comment]
> Thank you @lamby for the suggestions. I have not get the chance to work on
> the modifications yet. I would like to let you know that your feedback is clear to me
> and I will get back asap.

Brilliant stuff, looking forward to it :)

File: interface/frontend/index.html [hunk 7]
```diff
@@ -100,8 +100,8 @@ <h3 class="card-title">{{ selected.patient_id }}</h3>
     methods: {
       fetchData: function() {
         var vm = this;
-        this.$http.get('/api/images/').then(
-          function(response) {
+        this.$http.get('/api/images/available').then(
```
Comment (lamby) [inline interface/frontend/index.html]: Hm, I think this should use ``{% url … %}`` ?

File: interface/backend/api/views.py [hunk 8]
```diff
@@ -27,3 +28,14 @@ class NoduleViewSet(viewsets.ModelViewSet):
...
+
+class ImageAvailableApiView(APIView):
+    """
+    View list of images from dataset directory
+    """
+    def get(self, request):
+        """
+        Return a list of files and folders in dataset
+        """
+        pass
```
Comment (lamby) [inline interface/backend/api/views.py]: For now, should this return at least some kind of empty result? :)

File: interface/frontend/index.html (hunk 7 above)
Comment (rahit) [reply interface/frontend/index.html]: Whole <script> is under {% verbatim %}. Even if verbatim is omitted from script section, is it possible to use {% url %} inside Vue ?

**Comment by lamby** [comment]
Hey all, who is this blocking on? Whilst we are assigned to @rahit here, that may not be accurate AFICT; please update if this is wrong!  — lamby

**Comment by rahit** [comment]
@lamby  let me know if there is anything that should be done for this pull request.

**Comment by lamby** [comment]
@rahit Can you rebase against master; currently conflicting there :)

**Comment by rahit** [comment]
@lamby  done

**Comment by lamby** [comment]
Thanks, merged @rahit 
:)
